src/frontend/src/server.rs

```diff
@@ -143,7 +143,17 @@ where
         };
 
         let user_provider = self.plugins.get::<UserProviderRef>();
-        let runtime = match opts.mode {
+
+        // Determine whether it is Standalone or Distributed mode based on whether the meta client is configured.
+        let mode = {
+            if opts.meta_client.is_none() {
+                Mode::Standalone
+            } else {
+                Mode::Distributed
+            }
+        };
```

Comment:
non-blocking: We don't need the outer brackets.
```suggestion
        let mode = if opts.meta_client.is_none() {
            Mode::Standalone
        } else {
            Mode::Distributed
        };
```